tests/src/Functional/Provider/DecoratingCacheSettingsProviderTest.php

```diff
@@ -47,9 +62,40 @@ protected function createProvider(): SettingsProviderInterface
                 $container->get('test.settings_manager.serializer')
             ),
             $container->get('test.settings_manager.serializer'),
-            new FilesystemAdapter($namespace, 0, $cacheDir),
+            $this->cache,
             new Factory(new FlockStore()),
             0
         );
     }
+
+    /**
+     * @dataProvider dataProviderTestGetSettingsByName
+     */
+    public function testGetSettingsByName(array $domainNames, array $settingNames, array $expectedSettingsMap)
```

Comment:
Have you considered somehow testing call count to parent provider? Afterall this is the main reason behind the change and test does not check if it's actually working.

Reply:
yeah, good point. Will write a test